codebundles/aws-c7n-ebs-health/runbook.robot

```diff
@@ -0,0 +1,158 @@
+*** Settings ***
+Metadata          Author    saurabh3460
+Metadata          Support    AWS    EBS
+Documentation     Audit EBS resources by identifying unattached volumes, unused and aged snapshots, and unencrypted volumes.
+Force Tags    EBS    Volume    AWS    Storage    Secure
+
+Library    RW.Core
+Library    RW.CLI
+Library    CloudCustodian.Core
+
+Suite Setup    Suite Initialization
+
+
+
+*** Tasks ***
+List Unattached EBS Volumes in AWS Region `${AWS_REGION}` in AWS account `${AWS_ACCOUNT_ID}` 
+    [Documentation]  Check for unattached EBS volumes in the specified region. 
+    [Tags]    ebs    storage    aws    volume
+    ${c7n_output}=    RW.CLI.Run Cli
+    ...    cmd=custodian run -r ${AWS_REGION} --output-dir ${OUTPUT_DIR}/aws-c7n-ebs-health ${CURDIR}/unattached-ebs-volumes.yaml --cache-period 0
+    ...    secret__aws_account_id=${AWS_ACCESS_KEY_ID}
+    ...    secret__aws_secret_access_key=${AWS_SECRET_ACCESS_KEY}
+    ${report_data}=     RW.CLI.Run Cli
+    ...    cmd=cat ${OUTPUT_DIR}/aws-c7n-ebs-health/unattached-ebs-volumes/resources.json 
+    RW.Core.Add Pre To Report    ${c7n_output.stdout}     # Data needs to be parsed to be usable in the report.
+
+    ${parsed_results}=    CloudCustodian.Core.Parse EBS Results
+    ...    input_dir=${OUTPUT_DIR}/aws-c7n-ebs-health
+    RW.Core.Add Pre To Report    ${parsed_results} 
+
+    ${clean_output_dir}=    RW.CLI.Run Cli    cmd=rm -rf ${OUTPUT_DIR}/aws-c7n-ebs-health/unattached-ebs-volumes
+    # Convert custodian json output to a list.
+    TRY
+        ${ebs_volume_list}=    Evaluate    json.loads(r'''${report_data.stdout}''')    json
+        Log    ${report_data.stdout}
+    EXCEPT
+        Log    Failed to load JSON payload, defaulting to empty list.    WARN
+        ${ebs_volume_list}=    Create List
+    END
+
+    # Generate issues if any unused EBS volumes are in the list
+    IF    len(@{ebs_volume_list}) > 0
+        FOR    ${item}    IN    @{ebs_volume_list}
+            RW.Core.Add Issue        
+            ...    severity=2
+            ...    expected=EBS volumes in AWS Account `${AWS_ACCOUNT_NAME}` should be attached or in use
+            ...    actual=EBS volume `${item["VolumeId"]}` in AWS Account `${AWS_ACCOUNT_ID}` is unused
+            ...    title=Unused EBS volume `${item["VolumeId"]}` detected in AWS Account `${AWS_ACCOUNT_ID}`
+            ...    reproduce_hint=Review the volume details and usage in the AWS Management Console or CLI.
+            ...    details=${item}        # Include details such as volume ID, size, and region.
+            ...    next_steps="Escalate to service owner to review of unattached AWS EBS volume `${item["VolumeId"]}` in AWS Region `${AWS_REGION}` in AWS account `${AWS_ACCOUNT_ID}`. \n Delete unattached AWS EBS volume `${item["VolumeId"]}` in AWS Region `${AWS_REGION}` in AWS account `${AWS_ACCOUNT_ID}`."
```

Comment:
Remove specific volumeIds from the Next Steps - Next steps should look like another task in the workspace, which wouldn't be that specific. See the comment in your next task for further detail about using WHAT and WHERE in these tasks. Remove whitespace following `\n`

Reply:
can you please review this again made some changes. 

I haven't updated the next_steps according to `What` and `where`, and thinking to add this way. Let me know

`Unattached AWS EBS volume in AWS Region ${AWS_REGION} in AWS account ${AWS_ACCOUNT_ID} to escalate to the service owner for review.`